### ISSUE TITLE
Native Fetch types are seen as undefined types

### ISSUE THREAD

**Issue by DrakaSAN** [gajus/eslint-plugin-jsdoc#1097]
## Expected behavior
`RequestInfo` and `RequestInit` are recognized as built in types as they are provided and used by the native `fetch` from nodejs.

## Actual behavior

```
$ npx eslint ./test.js 

/path/to/test.js
  4:1  error  The type 'RequestInfo' is undefined  jsdoc/no-undefined-types
  5:1  error  The type 'RequestInit' is undefined  jsdoc/no-undefined-types

✖ 2 problems (2 errors, 0 warnings)
```

## ESLint Config

```json
{
	"root": true,
	"plugins": ["jsdoc"],
	"parserOptions": {
		"ecmaVersion": 2022,
		"sourceType": "script"
	},
	"env": {
		"node": true,
		"es2022": true
	},
	"rules": {
		"jsdoc/no-undefined-types": ["error"]
	},
	"settings": {
		"jsdoc": {
			"mode": "typescript"
		}
	}
}
```

## ESLint sample

```js
'use strict'

/**
 * @param {RequestInfo} url
 * @param {RequestInit | undefined} options
 * @returns {Promise<Response>}
 */
async function call(url, options) {
	return fetch(url, options)
}

module.exports = { call }
```

## Environment

- Node version: v18.16.0
- ESLint version v8.41.0
- `eslint-plugin-jsdoc` version: 44.2.7

**Comment by DrakaSAN** [comment]
Interestingly, while using eslint directly show this error, VSCode itself seems to not stumble on it
